homeassistant/components/plugwise/coordinator.py

```diff
@@ -104,24 +104,19 @@ def _async_remove_devices(self, data: PlugwiseData, entry: ConfigEntry) -> None:
         device_list = dr.async_entries_for_config_entry(
             device_reg, self.config_entry.entry_id
         )
-        # via_device cannot be None, this will result in the deletion
-        # of other Plugwise Gateways when present!
-        via_device: str = ""
-
         # First find the Plugwise via_device
-        for device_entry in device_list:
-            for identifier in device_entry.identifiers:
-                if identifier[0] != DOMAIN or identifier[1] != data.gateway[GATEWAY_ID]:
-                    continue
-                via_device = device_entry.id
-                break
+        gateway_device = device_reg.async_get_device(
+            {(DOMAIN, data.gateway[GATEWAY_ID])}
+        )
+        if gateway_device is not None:
+            via_device_id = gateway_device.id
```

Comment:
`via_device_id` can now be not set right?, doesn't that raise when it can't find it?

Reply:
If `gateway_device` is never `None`, I would suggest using an `assert` to make the code read as if we never expect it to be not found. (Currently the code reads as if that can be reality)

Reply:
Yeah, good point, the gateway should always be there.

Is this what you're suggesting?
```
        gateway_device = device_reg.async_get_device(
            {(DOMAIN, data.gateway[GATEWAY_ID])}
        )
        assert via_device_id = gateway_device.id
```

Reply:
Just for clarity - do we `assert` outside test-code or should we 'if None: raise SomeException'?

Reply:
I wouldn't mind removing the assert. The gateway is always there.

Reply:
I'd suggest we make it a `if gateway_device is None: raise PlugwiseException` then it will never trigger (i.e. noqa for testing)

Reply:
We use asserts if needed. I think this one is needed since it doesn't give the impression that there is another path the code can run.

Reply:
👍